bin/inference/pycbc_inference_model_stats

```diff
@@ -96,7 +96,10 @@ model.sampling_transforms = None
 def callmodel(arg):
     iteration, paramvals = arg
     # calculate the logposterior to get all stats populated
-    model.update(**{p: paramvals[p] for p in model.variable_params})
+    try:
```

Comment:
Again, this is not correct. Your top level model should just have an update method. That method should do whatever is needed to prepare for the log* methods to actually work. You shouldn't be requiring anyone know about a new method 'update_all_models'. It's fine if you want to have an new method for *internal* use, but not for the top-level api.

Reply:
So do I need to rename my `update_all_models` to be `update` (so overwrite the base one), or just use the original `update` in `callmodel`?

Reply:
@WuShichao A sampler is just using 'update' so why are you doing so differently here? Think through how to do this best, but it should be clear that making this change in this program seems very inconsistent. If it is required, that indicates something is wrong with your model, and if so fix that. Otherwise maybe you made this change in error.